moped-editor/src/views/projects/projectView/ProjectMilestones.js

```diff
@@ -28,14 +27,167 @@ import {
 import { useMutation } from "@apollo/client";
 import { format } from "date-fns";
 import parseISO from "date-fns/parseISO";
-import Autocomplete from "@mui/material/Autocomplete";
 
-// Helpers
 import { phaseNameLookup } from "src/utils/timelineTableHelpers";
-import DateFieldEditComponent from "./DateFieldEditComponent";
 import ToggleEditComponent from "./ToggleEditComponent";
-import ButtonDropdownMenu from "../../../components/ButtonDropdownMenu";
-import MilestoneTemplateModal from "./MilestoneTemplateModal";
+import MilestoneTemplateModal from "./ProjectMilestones/MilestoneTemplateModal";
+import MilestoneAutocompleteComponent from "./ProjectMilestones/MilestoneAutocompleteComponent";
+import DataGridDateFieldEdit from "./ProjectMilestones/DataGridDateFieldEdit";
+import DeleteConfirmationModal from "./DeleteConfirmationModal";
+
+const useMilestoneNameLookup = (data) =>
+  useMemo(() => {
+    if (!data) {
+      return {};
+    }
+    return data.moped_milestones.reduce(
+      (obj, item) =>
+        Object.assign(obj, {
+          [item.milestone_id]: item.milestone_name,
+        }),
+      {}
+    );
+  }, [data]);
+
+const useColumns = ({
+  data,
+  rowModesModel,
+  handleEditClick,
+  handleSaveClick,
+  handleCancelClick,
+  handleDeleteOpen,
+  milestoneNameLookup,
+}) =>
+  useMemo(() => {
+    return [
+      {
+        headerName: "Milestone",
```

Comment:
@chiaberry Could you update to autofocus this first columns when creating a new row?